plugins/filter/merge_data.py

```diff
@@ -0,0 +1,335 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2022     , DEMAREST Maxime <maxime@indelog.fr>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+name: merge_data
+short_description: Merge dict or lists by ensuring the comparing datas are
+                   presents, absents or identics.
+version_added: 5.0.2
+author: DEMAREST Maxime (@indelog)
+description:
+  - Merge two datas, current and expected in form of dict or list.
```

Comment:
```suggestion
  - Merge two data, current and expected in form of dict or list.
```
`data` is already plural, but I don't think you can say "two data" (https://www.merriam-webster.com/dictionary/data#usage-1). How about
```suggestion
  - Merge two data structures, current and expected in form of dict or list.
```
?